lib/routing/router.ts

```diff
@@ -1,50 +1,69 @@
-import { Routes, Route } from './routes';
+import { Route } from './routes';
 import { Middleware } from '../middleware/middleware';
 import { Controller } from '../controller/controller';
 import { GenericConstructor } from '../interfaces';
 import { Request } from '../request/request';
 import { Response } from '../response';
 import { Container } from 'inversify';
+import { HttpMethods } from '../request/event';
 
 
 export class Router<M extends Middleware, C extends Controller, R extends Route<M, C>> {
 
   private middleware: GenericConstructor<M>[] = [];
-  private subjectRoute: Route<M, C>;
+  private subjectRoute: R; 
+  private requestRoute: string;
+  private requestMethod: HttpMethods;
+  private pathParams: {};
 
   constructor(
     private request: Request,
     private container: Container
   ) { }
 
-  public route(routes: Routes<M, C, R>): void {
-    let requestRoute: string = this.request.getResource()
-    let requestMethod: string = this.request.getRequestMethod();
-    let routeGroup = routes.routes[requestRoute];
+  public route(routes: R[]): void {
+    this.requestRoute = this.request.getResource()
+    this.requestMethod = this.request.getRequestMethod();
 
-    this.addMiddlewareIfExists(routes.middleware);
-    this.addMiddlewareIfExists(routeGroup.middleware);
-
-    if (routeGroup[requestMethod] === undefined) {
-      // error route does not exists;
-      throw Error("Request method, '" + requestMethod
-        + ", does not exists on route, " + requestRoute + "'.");
-    }
-
-    this.subjectRoute = routeGroup[requestMethod];
+    try {
+      this.subjectRoute = this.getRequestRoute(routes);
+    } catch (e) {
+      throw e; // could not find route, lets just throw for now.
+    };
 
     this.addRouteMetaDataToRequest();
 
     this.addMiddlewareIfExists(this.subjectRoute.middleware);
   }
 
+  protected getRequestRoute(routes: R[]): R {
+    const isRequestedRoute = (route) => {
```

Comment:
Is there any reason you like doing this over inline? I don't really like it unless you pull it out into it's own private function on the class

Reply:
multiple reasons:
1) doing it inline instantiates 
2) cleaner not inline
3) private function would be okay but then you have to do 
```js
routes.find(route => this.isRequestedRoute(route));
//instead of
routes.find(isRequestedRoute)
```

Reply:
or I guess we could make it a private property?